spec/app/domain/authentication/constraints/non_permitted_constraint_spec.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+RSpec.describe(Authentication::Constraints::NonPermittedConstraint) do
+  let(:non_permitted_constraint) { ["iat", "nbf", "exp", "iss"] }
+  let(:raised_error) { ::Errors::Authentication::Constraints::NonPermittedRestrictionGiven }
+
+  subject(:constraint) do
+    Authentication::Constraints::NonPermittedConstraint.new(non_permitted: non_permitted_constraint)
+  end
+
+  context "when validating empty array" do
+    subject do
+      constraint.validate(resource_restrictions: [])
+    end
+
+    it "raises an error" do
+      expect { subject }.to_not raise_error
+    end
+  end
+
+  context "when validating one allowed restriction" do
+    subject do
+      constraint.validate(resource_restrictions: ["ref"])
+    end
+
+    it "raises an error" do
+      expect { subject }.to_not raise_error
+    end
+  end
+
+  context "when validating two allowed restriction" do
+    subject do
+      constraint.validate(resource_restrictions: ["ref", "sub"])
+    end
+
+    it "raises an error" do
+      expect { subject }.to_not raise_error
+    end
+  end
+
+  context "when validating one non permitted restriction" do
+    subject do
+      constraint.validate(resource_restrictions: ["exp"])
+    end
+
+    it "raises an error" do
+      expect { subject }.to raise_error(raised_error)
+    end
+  end
+
+  context "when validating two non permitted restrictions" do
+    subject do
+      constraint.validate(resource_restrictions: ["exp", "iat"])
+    end
+
+    it "raises an error" do
+      expect { subject }.to raise_error(raised_error)
+    end
+  end
+
+  context "when validating one permitted and one non permitted restriction" do
```

Comment:
Maybe worth adding - 
1. restriction with name that contains the non permited (customer made) - i.e. "expref"
2. Make the first restriction permitted and the second non-permitted 
3. create a list with first restriction permitted, second is empty string, third is non-permitted

Reply:
1. Name that contains inside something non permitted should be permitted in my opinion because there is no special treatment for strings that are different then the one in the const
2. Added it good point
3. This case is not possible because current load policy don't lets to load annitation without a name because its a broken yaml

Reply:
Regarding 1 i agree with you, we need to make sure that this is allowed and will be allowed also in future changes in the feature

Reply:
Ok now i got you. I will add it